tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+                if name[0] != "*" or name[-1] != "*":
+                    q.filter_name([name], GLOB)
+                else:
+                    q.filter_name([name.replace("*", "")], CONTAINS)
+            else:
+                q.filter_name([name], EQ)
+
+            if args.get("latest", False):
+                q.filter_latest_evr()
+
+            for package in list(q):
+                packages.append({
+                    "name": package.get_name(),
+                    "summary": package.get_summary(),
+                    "description": package.get_description(),
+                    "url": package.get_url(),
+                    "repo_id": package.get_repo_id(),
+                    "epoch": int(package.get_epoch()),
+                    "version": package.get_version(),
+                    "release": package.get_release(),
+                    "arch": package.get_arch(),
+                    "buildtime": self._timestamp_to_rfc3339(package.get_build_time()),
+                    "license": package.get_license()
+                })
+        return packages
+
+    def depsolve(self, transactions):
+        """depsolve returns a list of the dependencies for the set of transactions
+        """
+        last_transaction = []
+
+        for transaction in transactions:
+            goal = dnf5.base.Goal(self.base)
+            goal.reset()
+            sack = self.base.get_rpm_package_sack()
+            sack.clear_user_excludes()
+
+            # weak deps are selected per-transaction
+            self.base.get_config().install_weak_deps = transaction.get("install_weak_deps", False)
+
+            # set the packages from the last transaction as installed
+            for installed_pkg in last_transaction:
+                goal.add_rpm_install(installed_pkg)
+
+            # Support group/environment names as well as ids
+            settings = dnf5.base.GoalJobSettings()
+            settings.group_with_name = True
+
+            # Packages are added individually, excludes are handled in the repo setup
+            for pkg in transaction.get("package-specs"):
+                goal.add_install(pkg, settings)
+            transaction = goal.resolve()
+            if transaction.get_problems() != NO_PROBLEM:
+                raise TransactionError("\n".join(transaction.get_resolve_logs_as_strings()))
+
+            # store the current transaction result
+            last_transaction.clear()
+            for tsi in transaction.get_transaction_packages():
+                # Only add packages being installed, upgraded, downgraded, or reinstalled
+                if not dnf5.base.transaction.transaction_item_action_is_inbound(tsi.get_action()):
+                    continue
+                last_transaction.append(tsi.get_package())
+
+        # Something went wrong, but no error was generated by goal.resolve()
+        if len(transactions) > 0 and len(last_transaction) == 0:
+            raise TransactionError("Empty transaction results")
```

Comment:
Would it make sense to include some context here, like `Empty transaction result for ...` or is this clear from the traceback already (pardon again my ignorance here)?

Reply:
I'm not sure what would help. You already know what you fed into it, so repeating that won't tell you anything more.